### PR TITLE
46471 - Make rollingUpdate configurable in istio/gateway deployment template

### PR DESCRIPTION
**Pull request by hari-rad**
Just to add configurable values for rollingUpdate.maxSurge and rollingUpdate.maxUnvailable in deployment template for istio/gateway.

### CODE REVIEW

**Comment by zirain** [comment]
please sign CLA, @hari-rad .

File: manifests/charts/gateway/values.yaml [hunk 1]
```diff
@@ -129,3 +129,8 @@ imagePullSecrets: []
 podDisruptionBudget: {}
 
 terminationGracePeriodSeconds: 30
+
+# Use RollingUpdate deployment strategy to reduce disruption
+rollingUpdate:
+  maxSurge: 50%
+  maxUnavailable: 25%
```
Comment (jacob-delgado) [inline manifests/charts/gateway/values.yaml:136]: istio tends to take a bit more agnostic approach to this and would prefer admins/operations to make the determination on what to set these values to. Can you templatize this similar to `podDisruptionBudget`?

**Comment by dawid-remitly** [comment]
Hi @hari-rad will you continue this? I think jacob is asking to set this to more generic `strategy: {}` setup

**Pull request closed without merging**